rust/connlib/tunnel/src/device_channel.rs

```diff
@@ -70,7 +71,27 @@ impl Device {
         }
     }
 
-    #[cfg(any(target_os = "android", target_os = "linux"))]
+    pub(crate) fn set_tun(&mut self, tun: Tun) {
+        self.tun = Some(tun);
+
+        if let Some(waker) = self.waker.take() {
+            waker.wake();
+        }
+    }
+
+    #[cfg(target_os = "android")]
+    pub(crate) fn set_config(
```

Comment:
The implementations of `set_config` have diverged a lot, but I'm not sure if that should be documented or refactored.

I see here that on Android it calls `on_set_interface_config`, but on Linux it sets the tunnel instead. They seem totally different.

Reply:
Yes, they are diverging now until we unify where we handle the route updates. I've described this in more detail in #4473. The end-goal is for connlib on only ever call `on_set_interface_config` (or later emit an event). On Android and Darwin, we already do that today, previously we just did it inside `Tun` instead of out here. If you look at `tun_android.rs`, it doesn't actually set the routes but instead just calls the callback, same for Darwin.